### PR TITLE
Fix: missing dependency spring-data-commons

### LINKED ISSUE #15040: Gateway without database missing dependency spring-data-commons
##### **Overview of the issue**

[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /home/dej/projects/redfroggy/jhi/gateway/src/main/java/fr/redfroggy/jhi/config/WebConfigurer.java:[9,36] package org.springframework.data.web does not exist
[ERROR] /home/dej/projects/redfroggy/jhi/gateway/src/main/java/fr/redfroggy/jhi/config/WebConfigurer.java:[10,36] package org.springframework.data.web does not exist
[INFO] 2 errors 
[INFO] -------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE

##### **Motivation for or Use Case**

missing dependency spring-data-commons

##### **Reproduce the error**

run jhipster with provided .yo-rc.json 

##### **Suggest a Fix**

add in pom.xml
```
        <dependency>
            <groupId>org.springframework.data</groupId>
            <artifactId>spring-data-commons</artifactId>
        </dependency>
```

##### **JHipster Version(s)**

```
gateway@0.0.0 /home/dej/projects/redfroggy/jhi/gateway
└── generator-jhipster@7.0.1 

```

##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**

<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "blueprints": [],
    "otherModules": [],
    "applicationType": "gateway",
    "baseName": "gateway",
    "jhipsterVersion": "7.0.1",
    "skipClient": true,
    "skipServer": false,
    "skipUserManagement": true,
    "skipCheckLengthOfIdentifier": false,
    "skipFakeData": false,
    "jhiPrefix": "jhi",
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "testFrameworks": ["cucumber"],
    "pages": [],
    "creationTimestamp": 1621346430684,
    "serviceDiscoveryType": "eureka",
    "reactive": true,
    "authenticationType": "oauth2",
    "packageName": "fr.redfroggy.jhi",
    "serverPort": "8080",
    "cacheProvider": "no",
    "enableHibernateCache": false,
    "databaseType": "no",
    "buildTool": "maven",
    "serverSideOptions": [],
    "websocket": false,
    "searchEngine": false,
    "messageBroker": false,
    "enableSwaggerCodegen": false,
    "clientFramework": "no",
    "enableTranslation": false,
    "nativeLanguage": "fr",
    "packageFolder": "fr/redfroggy/jhi",
    "devDatabaseType": "no",
    "prodDatabaseType": "no",
    "clientPackageManager": "npm",
    "clientTheme": "none",
    "clientThemeVariant": "",
    "withAdminUi": true,
    "languages": ["en", "fr"]
  }
}

</pre>
</details>

##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>

</pre>
</details>

##### **Environment and Tools**

openjdk version "11.0.11" 2021-04-20
OpenJDK Runtime Environment (build 11.0.11+9-Ubuntu-0ubuntu2.20.04)
OpenJDK 64-Bit Server VM (build 11.0.11+9-Ubuntu-0ubuntu2.20.04, mixed mode, sharing)

git version 2.25.1

node: v14.15.1

npm: 6.14.9

Docker version 19.03.13, build cd8016b6bc

docker-compose version 1.25.5, build unknown

No change to package.json was detected. No package manager install will be executed.

##### **Browsers and Operating System**

- [x] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by qmonmert**
Fix #15040 

---

Please make sure the below checklist is followed for Pull Requests.

- [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [x] Tests are added where necessary
- [x] The JDL part is updated if necessary
- [x] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [x] Documentation is added/updated where necessary
- [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/server/templates/pom.xml.ejs [hunk 1]
```diff
@@ -644,6 +644,10 @@
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-validation</artifactId>
         </dependency>
+        <dependency>
```
Comment (pascalgrimaud) [inline generators/server/templates/pom.xml.ejs:647]: do we need this for default configuration ? Or is it specific to gateway without database ?

Comment (qmonmert) [reply generators/server/templates/pom.xml.ejs:647]: it seems useful for all reactive projects

**Review by atomfrede — COMMENTED** [review]
Do we need to add it to build.gradle as well?

**Comment by qmonmert** [comment]
@atomfrede I guess yes but I do not feel comfortable with Gradle...

**Comment by atomfrede** [comment]
I will check.

**Comment by pascalgrimaud** [comment]
Gradle part can be done separately

**Pull request merged**